core/src/main/java/fr/sncf/osrd/envelope_sim/EnvelopePath.java

```diff
@@ -64,13 +64,38 @@ private double getCumGrade(double position) {
         return gradeCumSum[gradeRangeIndex] + gradeValues[gradeRangeIndex] * (position - gradeRangeStart);
     }
 
+
+    /** For a given position, return the index of the position just before in gradePositions */
+    public int getIndexBeforePos(double position) {
+        assert position <= gradePositions[gradePositions.length - 1] && position >= gradePositions[0];
+        for (int i = 0; i < gradePositions.length; i++) {
+            var pos = gradePositions[i];
+            if (pos > position)
+                return i - 1;
+        }
+        return gradePositions.length - 1;
+    }
+
     @Override
     public double getAverageGrade(double begin, double end) {
         if (begin == end)
             return getCumGrade(begin);
         return (getCumGrade(end) - getCumGrade(begin)) / (end - begin);
     }
 
+    @Override
+    public double getLowestGrade(double begin, double end) {
```

Comment:
This method returns the "largest" negative grade, it seems to be the intended behavior but I'd prefer if we have a comment that explains it (both here and in the api file). I'd expect a function named `getLowest` to return something close to 0